### PR TITLE
Set return values in emsmdbp_object_get_properties_mapistore_root

### PR DESCRIPTION
**Pull request by icandigitbaby**
The array of return values wasn't being updated for all the properties.
This, plus the lack of sanity checks, was leading to crashes when there
was an error retrieving a property.

### CODE REVIEW

File: mapiproxy/servers/default/emsmdb/emsmdbp_object.c [hunk 1]
```diff
@@ -3035,28 +3036,49 @@ static int emsmdbp_object_get_properties_mapistore_root(TALLOC_CTX *mem_ctx, str
 			}
 		}
 		else if (properties->aulPropTag[i] == PR_CONTENT_COUNT) {
-                        /* a hack to avoid fetching dynamic fields from openchange.ldb */
-                        obj_count = talloc_zero(data_pointers, uint32_t);
+			/* a hack to avoid fetching dynamic fields from openchange.ldb */
```
Comment (sixstone-qq) [inline mapiproxy/servers/default/emsmdb/emsmdbp_object.c]: This comment is out of date, right?

File: mapiproxy/servers/default/emsmdb/emsmdbp_object.c [hunk 2]
```diff
@@ -3126,9 +3153,9 @@ static int emsmdbp_object_get_properties_mapistore_root(TALLOC_CTX *mem_ctx, str
 				properties->aulPropTag[i] = PidTagCreationTime;
 			}
 			retval = openchangedb_get_folder_property(data_pointers, emsmdbp_ctx->oc_ctx, owner, properties->aulPropTag[i], folder->folderID, data_pointers + i);
-                }
+		}
 		retvals[i] = retval;
-        }
+    }
 
 	return MAPISTORE_SUCCESS;
```
Comment (sixstone-qq) [inline mapiproxy/servers/default/emsmdb/emsmdbp_object.c]: Here the indentation fails with the latest closing curly brace.

**Comment by icandigitbaby** [comment]
Thanks for the review. The (indeed) outdated comment was removed and the indentation was corrected.

**Comment by blaxter** [comment]
Why `mapistore_folder_get_child_count`, `mapistore_folder_get_child_count`, `emsmdbp_folder_get_folder_count`, `emsmdbp_source_key_from_fmid`, `mapistore_properties_get_properties` are returning success if the data was not found. My concern is that we are doing here the work that needs to be implemented in each function.

And actually I think only a few of them could return `NULL` data (and/or bad return code = SUCCESS and NULL data)

**Comment by icandigitbaby** [comment]
I shared the same concerns, I kept those double-checks after discussing it with @sixstone-qq and deciding that being paranoid is acceptable on this part of the code. But I agree that it reduces legibility and we should be able to rely on the functions we call, so I'm OK with removing these checks.

**Comment by sixstone-qq** [comment]
:+1:

**Comment by blaxter** [comment]
:ok_hand:

**Pull request merged**